app/helpers/application_helper.rb

```diff
@@ -3,33 +3,30 @@ module ApplicationHelper
   include HomeHelper
 
   # Add autogenerated html data-id attribute if not present (prefixed with "aid_")
-  def link_to(*args, &block)
-    if block_given?
-      super(*args)
-    else
-      options = args[1] || {}
-      html_options = args[2] || {}
-      unless html_options.has_key?(:'data-id') || (options.is_a?(String) && options.starts_with?("mailto:"))
-        begin
-          path = URI.split(url_for(options) || html_options['href'])[5].split(/\//).select {|x| !x.empty?}
-          if !path.empty?
-            max = (path.size <= 3) ? path.size : 3
-            id = path.last(max).join('_')
-          else
-            id = 'not_defined'
-          end
-        rescue => e
-          Foreman::Logging.exception("Failed generating link using #{args.inspect}", e)
-          id = 'not_parseable'
-        end
-        html_options[:'data-id'] = "aid_#{id}"
+  def link_to(*args)
+    return super(*args) if block_given?
+
+    options = args[1] || {}
+    html_options = args[2] || {}
+    unless html_options.has_key?(:'data-id') || (options.is_a?(String) && options.starts_with?("mailto:"))
+      begin
+        path = URI(url_for(options) || html_options['href']).path.split('/').reject(&:empty?)
+        id = if path.empty?
+               'not_defined'
+             else
+               path.last([path.size, 3].max).join('_')
+             end
+      rescue => e
+        Foreman::Logging.exception("Failed generating link using #{args.inspect}", e)
+        id = 'not_parseable'
       end
-      if html_options[:confirm]
-        html_options[:data] ||= {}
-        html_options[:data][:confirm] = html_options.delete(:confirm)
-      end
-      super(args[0], args[1], html_options)
+      html_options[:'data-id'] = "aid_#{id}"
```

Comment:
Without looking at the details, should this even set `data-id` if id is `not_parseable` or `not_defined`? How useful is `aid_not_parseable`? Will anything ever query this and what happens if multiple elements are not parseable?

Reply:
I'm not sure yet what is the usage of `data-id`, but I agree with you that it is a good thing to check if `id` actually present, and exists, changing that